framework/src/loops/MaxQpsThread.C

```diff
@@ -64,19 +64,41 @@ MaxQpsThread::operator()(const ConstElemRange & range)
     // We'll use one for element interiors, which calculates nothing
     std::unique_ptr<FEBase> fe(FEBase::build(dim, fe_type));
     fe->get_nothing();
+    // Alex: This seems wrong, e.g. like we'd want the quadrature order to be higher if the
+    // polynomial basis is higher, but our fe_type up above is first Lagrange, so it seems like the
+    // person who wrote this code didn't want to consider higher order bases even if if the user
+    // asked for higher order bases
+    fe->add_p_level_in_reinit(false);
 
     // And another for element sides, which calculates the minimum
     // libMesh currently allows for that
     std::unique_ptr<FEBase> side_fe(FEBase::build(dim, fe_type));
     side_fe->get_xyz();
+    side_fe->add_p_level_in_reinit(false);
 
     // figure out the number of qps for volume
     auto qrule = assem.attachQRuleElem(dim, *fe);
     fe->reinit(elem);
     if (qrule->n_points() > _max)
       _max = qrule->n_points();
 
-    unsigned int n_shape_funcs = fe->n_shape_functions();
+    const auto n_shape_funcs = [dim, &fe]()
+    {
+      const auto elem_type = fe->get_type();
+      switch (dim)
+      {
+        case 0:
+          return static_cast<FE<0, LAGRANGE> *>(fe.get())->n_shape_functions(elem_type, FIRST);
```

Comment:
And what's going on here?  While figuring out our maximum number of quadrature points (useful for pre-sizing working arrays), we also figure out the maximum number of shape functions a first-order LAGRANGE variable would have (useful for writing code that will break if ever asked to use non-trivial FEM?) too?

Could we just mark `MaxQpsThread::max_shape_funcs()` and `FEProblemBase::getMaxShapeFunctions()` with a `mooseError` and see if anything in CI needs to be fixed?  I'm not seeing any internal usage.

Reply:
I just went ahead and deleted it and CI seems happy enough